src/bika/lims/content/abstractanalysis.py

```diff
@@ -596,7 +596,7 @@ def calculateResult(self, override=False, cascade=False):
 
         # convert any remaining placeholders, e.g. from interims etc.
         # NOTE: we assume remaining values are all floatable!
-        formula = formula.replace("[", "%(").replace("]", ")f")
+        formula = formula.replace("[", "%(").replace("]", ")r")
```

Comment:
Note that this will eventually convert any value passed in through a placeholder (e.g. `[most_1]` becomes `%(most_1)r`) using `repr()` (kind-of *official* string representation of an object). For some types, the conversion using `str` (kind-of *informal* string representation of an object) does not match with the conversion using `repr` (e.g. `datetime`).

Since we know the type of an interim beforehand (`result_type` key), I suggest to not change this line, but apply the proper converter for each interim before, at https://github.com/senaite/senaite.core/pull/2476/files#diff-85f0e866b6d6166de01fe6d6f109fe28ed787c8b4e51980a98efb2c256df70a3R557) where we do have both the keyword of the parameter/interim (e.g. `most_1` or `most_2`) and we can easily get the type with `interim.get("result_type")`. 

Basically I would expect a interim with a `result_type` value other than empty (see https://github.com/senaite/senaite.core/blob/2.x/src/bika/lims/browser/fields/interimfieldsfield.py#L101) to be converted as follows:

```python
# Convert 'Numeric' interim values using `float`. Convert the rest using `str`
converter = "s" if interim.get("result_type") else "f"
formula = formula.replace("[" + keyword + "]", "%(" + keyword + ")" + converter)
```